src/main/java/calculator/domain/Calculator.java

```diff
@@ -0,0 +1,43 @@
+package calculator.domain;
+
+import calculator.validator.InputValidator;
+import calculator.validator.NegativeNumberValidator;
+import calculator.util.SeparatorExtractor;
+
+public class Calculator {
+
+    public int add(String input) {
+        if (isInputEmpty(input)) {
+            return 0;
+        }
+
+        String separator = SeparatorExtractor.extractSeparators(input);
+        input = SeparatorExtractor.extractNumbersPart(input);
+
+        InputValidator.validateInput(input, separator);
+
+        String[] numbers = splitNumbers(input, separator);
+        return calculateSum(numbers);
+    }
+
+    private boolean isInputEmpty(String input) {
+        return input == null || input.isEmpty();
+    }
+
+    private String[] splitNumbers(String input, String separator) {
+        for (char tempSeparator : separator.toCharArray()) {
+            input = input.replace(String.valueOf(tempSeparator), " ");
+        }
+        return input.split("\\s+");
```

Comment:
정규식을 사용하면 효율적일 것 같아요
```suggestion
       String regex = "[" + separators + "]";
       return input.split(regex);
```